### PR TITLE
Required picasso information

### LINKED ISSUE #3179: Add required information to picasso printing
J+S requires some things to be presented directly on the picasso which is submitted to J+S coaches and in the NDS. We should probably make sure these are always displayed on the picasso when printing, at least for now, as long as there is only one non-configurable way to print the picasso. For legal reasons I can't link to the brochure (Lagersport/Trekking Lager) where this defined, but I hope it is okay to list the requirements here. The brochure is available digitally to J+S leaders [here](https://www.jugendundsport.ch/content/jus-internet/de/sportarten/lagersport-trekking-uebersicht/login-leiter.html).

## J+S Camps
Bestandteile des Grobprogramms als Wochenraster
- [ ] Organisator (Name der Jugendorganisation).
- [ ] Lagerart (Haus-, Zelt-, Unterwegslager, Sommer-, Herbstlager).
- [ ] Lagerdaten (inkl. Jahr).
- [ ] Ort und Name des Lagerhauses bzw. Lagerplatzes.
- [ ] Name(n) der Lagerleitung.
- [ ] Name des J+S-Coachs.
- [x] Tag, Datum und Tagesverantwortliche.
- [x] Titel jeder Aktivität und verantwortliche Person.
- [x] J+S-Aktivitäten als LS oder LA gekennzeichnet.
- [x] An- und Rückreisezeit.
- [ ] Lagereinkleidung

Users can theoretically add the missing information to the picasso e.g. as activities on a first camp day. But that is not a very elegant solution for a requirement which concerns most of the real camps on eCamp.

## J+S Courses
Source: [Anker](https://pfadi.swiss/media/files/4e/301318de-anker-20220816-akom.pdf) chapter 4.1
> [Es] müssen für BSV und J+S ersichtlich sein:

- [x] Daten
- [ ] Kursnummer
- [ ] Kursbezeichnung (bei J+S Kurs: J+S Bezeichnung)
- [ ] Kursort
- [ ] bürgerlicher Name von HKL und LKB
- [ ] J+S-Logo (bei J+S-Kurs)

### PR DESCRIPTION
**Pull request by carlobeltrame**
Fixes #3179

### CODE REVIEW

File: api/src/Entity/CampCollaboration.php [hunk 1]
```diff
@@ -86,7 +86,7 @@
 #[ORM\Index(columns: ['status'])]
 class CampCollaboration extends BaseEntity implements BelongsToCampInterface {
     public const ITEM_NORMALIZATION_CONTEXT = [
-        'groups' => ['read', 'CampCollaboration:Camp', 'CampCollaboration:User'],
+        'groups' => ['read', 'CampCollaboration:Camp', 'CampCollaboration:User', 'User:Profile'],
```
Comment (carlobeltrame) [inline api/src/Entity/CampCollaboration.php]: Revert this and create a filter by camp or collaboration or user on the profiles endpoint

File: api/src/Entity/Camp.php [hunk 2]
```diff
@@ -65,7 +65,7 @@
 #[ORM\UniqueConstraint(name: 'couponKey_unique', columns: ['couponKey'])]
 class Camp extends BaseEntity implements BelongsToCampInterface, CopyFromPrototypeInterface {
     public const ITEM_NORMALIZATION_CONTEXT = [
-        'groups' => ['read', 'Camp:Periods', 'Period:Days', 'Camp:CampCollaborations', 'CampCollaboration:User'],
+        'groups' => ['read', 'Camp:Periods', 'Period:Days', 'Camp:CampCollaborations', 'CampCollaboration:User', 'User:Profile'],
```
Comment (carlobeltrame) [inline api/src/Entity/Camp.php]: Revert this and create a filter by camp or collaboration or user on the profiles endpoint

File: frontend/src/components/print/print-react/components/picasso/Picasso.jsx [hunk 3]
```diff
@@ -56,16 +59,28 @@ function Picasso(props) {
...
+              title: props.$tc('print.picasso.title', { period: period.description }),
+              fit: true,
+            }}
+            style={{ ...styles.h1, ...picassoStyles.title }}
+          >
+            {props.$tc('print.picasso.title', { period: period.description })}
+          </Text>
+          <Text>{period.camp().organizer}</Text>
+          {period.camp().printYSLogoOnPicasso ? (
+            <YSLogo
```
Comment (carlobeltrame) [inline frontend/src/components/print/print-react/components/picasso/Picasso.jsx:75]: There is a z-index issue, the white border of the table below is visible on the logo sometimes

File: print/components/PicassoChunk.vue [hunk 4]
```diff
@@ -138,6 +243,14 @@ export default {
   transform: rotate(-90deg);
   position: relative;
   top: 320px;
+
+  .fullwidth {
+    width: 960px;
```
Comment (carlobeltrame) [inline print/components/PicassoChunk.vue:248]: Print preview looks fine, but it breaks in the printed pdf (maybe due to the ecamp3 header and page number footer). Maybe try to reduce this width a little. Soon we should be able to remove the rotate 90deg hack, when the browserless chrome supports switching to landscape mode.

File: frontend/src/components/print/print-react/components/picasso/Categories.jsx [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+import { Text, View } from '@react-pdf/renderer'
+import picassoStyles from './picassoStyles.js'
+import CategoryLabel from '../CategoryLabel.jsx'
+
+function Categories({ period }) {
+  const camp = period.camp()
+  return (
+    <View style={picassoStyles.categories}>
+      {camp.categories().items.map((category) => (
+        <View style={picassoStyles.category}>
```
Comment (BacLuc) [inline frontend/src/components/print/print-react/components/picasso/Categories.jsx]: don't you need a key property here?

**Review by BacLuc — APPROVED** [review]

File: frontend/src/components/print/print-react/components/picasso/Categories.jsx (hunk 5 above)
Comment (carlobeltrame) [reply frontend/src/components/print/print-react/components/picasso/Categories.jsx]: Yes, thanks

File: print/components/PicassoChunk.vue (hunk 4 above)
Comment (carlobeltrame) [reply print/components/PicassoChunk.vue:248]: I "fixed" it for now by reducing the page margins in the @page CSS from 15mm to 10mm.

**Comment by pmattmann** [comment]
Landscape is super. And if we wish to merge fast - feel free.
Please check Picasso-Print in Portrait as well.

Nuxt:  Page-Break
Screenshot 2023-04-07 110749

React:  Block is missing
Screenshot 2023-04-07 110851

**Review by pmattmann — APPROVED** [review]

**Comment by carlobeltrame** [comment]
> Nuxt: Page-Break

I adjusted the picasso height on portrait mode, so your example now should look okay. But since the vuetify calendar has a fixed height, for now it does not adjust at all depending on the rest of the page content. So in camps with two rows of categories, or camps with very long leader lists or addresses, the layout will still break. This will need a general overhaul, probably by @usu some time. For now I hope it is good enough.
 
> React: Block is missing

I found a bug, this schedule entry was always squished into zero height at the bottom of the day. Thanks for mentioning it! This happened when a schedule entry started within the last hour of the day, and it will become all the more important with a follow-up PR which I'm working on, where I plan to auto-detect the best "end of day" hour (instead of hard-coding it to 24:00)

**Pull request merged**